components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIManagerConfiguration.java

```diff
@@ -626,6 +523,120 @@ private void readChildElements(OMElement serverConfig,
+            environment.setDescription(description.getText());
+        } else {
+            environment.setDescription("");
+        }
+        environment.setReadOnly(true);
+        List<VHost> vhosts = new LinkedList<>();
+        environment.setVhosts(vhosts);
+        environment.setEndpointsAsVhost();
+        Iterator vhostIterator = environmentElem.getFirstChildWithName(new QName(
+                APIConstants.API_GATEWAY_VIRTUAL_HOSTS)).getChildrenWithLocalName(
+                APIConstants.API_GATEWAY_VIRTUAL_HOST);
+        while (vhostIterator.hasNext()) {
+            OMElement vhostElem = (OMElement) vhostIterator.next();
+            String httpEp = APIUtil.replaceSystemProperty(vhostElem.getFirstChildWithName(new QName(
+                    APIConstants.API_GATEWAY_VIRTUAL_HOST_HTTP_ENDPOINT)).getText());
+            String httpsEp = APIUtil.replaceSystemProperty(vhostElem.getFirstChildWithName(new QName(
+                    APIConstants.API_GATEWAY_VIRTUAL_HOST_HTTPS_ENDPOINT)).getText());
+            String wsEp = APIUtil.replaceSystemProperty(vhostElem.getFirstChildWithName(new QName(
+                    APIConstants.API_GATEWAY_VIRTUAL_HOST_WS_ENDPOINT)).getText());
+            String wssEp = APIUtil.replaceSystemProperty(vhostElem.getFirstChildWithName(new QName(
+                    APIConstants.API_GATEWAY_VIRTUAL_HOST_WSS_ENDPOINT)).getText());
+            String webSubHttpEp = APIUtil.replaceSystemProperty(vhostElem.getFirstChildWithName(new QName(
+                    APIConstants.API_GATEWAY_VIRTUAL_HOST_WEBSUB_HTTP_ENDPOINT)).getText());
+            String webSubHttpsEp = APIUtil.replaceSystemProperty(vhostElem.getFirstChildWithName(new QName(
+                    APIConstants.API_GATEWAY_VIRTUAL_HOST_WEBSUB_HTTPS_ENDPOINT)).getText());
+
+            //Prefix websub endpoints with 'websub_' so that the endpoint URL
+            // would begin with: 'websub_http://', since API type is identified by the URL protocol below.
+            webSubHttpEp = "websub_" + webSubHttpEp;
+            webSubHttpsEp = "websub_" + webSubHttpsEp;
+
+            VHost vhost = VHost.fromEndpointUrls(new String[]{
+                    httpEp, httpsEp, wsEp, wssEp, webSubHttpEp, webSubHttpsEp});
+            vhosts.add(vhost);
+        }
+        OMElement properties = environmentElem.getFirstChildWithName(new
+                QName(APIConstants.API_GATEWAY_ADDITIONAL_PROPERTIES));
+        Map<String, String> additionalProperties = new HashMap<>();
+        if (properties != null) {
+            Iterator gatewayAdditionalProperties = properties.getChildrenWithLocalName
+                    (APIConstants.API_GATEWAY_ADDITIONAL_PROPERTY);
+            while (gatewayAdditionalProperties.hasNext()) {
+                OMElement propertyElem = (OMElement) gatewayAdditionalProperties.next();
+                String propName = propertyElem.getAttributeValue(new QName("name"));
+                String propValue = propertyElem.getText();
```

Comment:
check the value from vault same as https://github.com/wso2/carbon-apimgt/pull/10839/files#diff-6ad5a30e0da551268318842e70c7a6e235b4e71a5985b57b662723142c8d9ff9R556-R557

Reply:
Fixed with 8bcaf927123b460db95dcb2223becbdbff7946eb